File: src/main/java/com/github/nsc/de/shake/parser/node/logical/LogicalNode.java
package com.github.nsc.de.shake.parser.node.logical;

import com.github.nsc.de.shake.lexer.characterinput.position.PositionMap;
import com.github.nsc.de.shake.parser.node.ValuedNode;

public abstract class LogicalNode extends ValuedNode {

    public LogicalNode(PositionMap map) {
        super(map);
    }

    abstract String getOperator();

}

File: parser/src/commonMain/kotlin/com/github/shakelang/shake/parser/node/logical/LogicalNode.kt
package com.github.shakelang.shake.parser.node.logical

import com.github.shakelang.shake.parser.node.ValuedNode
import com.github.shakelang.shake.parser.node.logical.LogicalConcatenationNode
import com.github.shakelang.shake.parser.node.logical.LogicalCompareNode
import com.github.shakelang.shake.parser.node.logical.LogicalNode
import com.github.shakelang.shake.util.characterinput.position.PositionMap

abstract class LogicalNode(map: PositionMap) : ValuedNode(map) {
    abstract val operator: String?
}